src/OpenLoco/src/Map/MapGenerator/MapGenerator.cpp

```diff
@@ -433,13 +435,108 @@ namespace OpenLoco::World::MapGenerator
         }
     }
 
+    static void updateProgress(uint8_t value);
+
+    // 0x004595B7
+    static bool isCargoProducedAnywhere(uint8_t requiredCargoType)
+    {
+        auto commonBuildingCargoType = IndustryManager::getMostCommonBuildingCargoType();
+        if (commonBuildingCargoType == requiredCargoType)
+            return true;
+
+        for (auto i = 0U; i < ObjectManager::getMaxObjects(ObjectType::industry); i++)
+        {
+            auto* industryObj = ObjectManager::get<IndustryObject>(i);
+            if (industryObj == nullptr)
+            {
+                continue;
+            }
+
+            if (getCurrentYear() < industryObj->designedYear || getCurrentYear() > industryObj->obsoleteYear)
+            {
+                continue;
+            }
+
+            for (auto producedCargoType : industryObj->producedCargoType)
+            {
+                if (producedCargoType == requiredCargoType)
+                    return true;
+            }
+        }
+
+        return false;
+    }
+
     // 0x004597FD
     static void generateIndustries(uint32_t minProgress, uint32_t maxProgress)
     {
-        registers regs;
-        regs.eax = minProgress;
-        regs.ebx = maxProgress;
-        call(0x004597FD, regs);
+        auto numIndustriesAvailable = 0;
+        for (auto indObjId = 0U; indObjId < ObjectManager::getMaxObjects(ObjectType::industry); indObjId++)
+        {
+            if (ObjectManager::get<IndustryObject>(indObjId) != nullptr)
+            {
+                numIndustriesAvailable++;
+            }
+        }
+
+        if (!numIndustriesAvailable)
```

Comment:
i wish there was a compiler option to disable implicit int-bool conversions like this. personally i'd write `if (numIndustriesAvailable == 0)`

Reply:
Heh, I used an explicit comparison elsewhere as well. I'll change it here, too, for consistency.